src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
         }
 
         public override async Task<IQueueEntry<T>> DequeueAsync(TimeSpan? timeout = null) {
+            await EnsureQueueCreatedAsync().AnyContext();
+
             using (var msg = await _queueClient.ReceiveAsync(timeout ?? TimeSpan.FromSeconds(30)).AnyContext()) {
                 return await HandleDequeueAsync(msg).AnyContext();
             }
         }
 
-        public override Task<IQueueEntry<T>> DequeueAsync(CancellationToken cancellationToken) {
+        protected override Task<IQueueEntry<T>> DequeueImplAsync(CancellationToken cancellationToken) {
             _logger.Warn("Azure Service Bus does not support CancellationTokens - use TimeSpan overload instead. Using default 30 second timeout.");
 
             return DequeueAsync();
         }
 
         public override async Task RenewLockAsync(IQueueEntry<T> entry) {
+            await EnsureQueueCreatedAsync().AnyContext(); // Azure SB needs to call this as it populates the _queueClient field
+
             await _queueClient.RenewMessageLockAsync(new Guid(entry.Id)).AnyContext();
             await OnLockRenewedAsync(entry).AnyContext();
         }
 
         public override async Task CompleteAsync(IQueueEntry<T> entry) {
+            await EnsureQueueCreatedAsync().AnyContext(); // Azure SB needs to call this as it populates the _queueClient field
+
             Interlocked.Increment(ref _completedCount);
             await _queueClient.CompleteAsync(new Guid(entry.Id)).AnyContext();
             await OnCompletedAsync(entry).AnyContext();
         }
         
         public override async Task AbandonAsync(IQueueEntry<T> entry) {
+            await EnsureQueueCreatedAsync().AnyContext(); // Azure SB needs to call this as it populates the _queueClient field
+
             Interlocked.Increment(ref _abandonedCount);
             await _queueClient.AbandonAsync(new Guid(entry.Id)).AnyContext();
             await OnAbandonedAsync(entry).AnyContext();
         }
         
         public override void Dispose() {
-            _queueClient.Close();
+            _queueClient?.Close();
```

Comment:
This probably also needs to call dispose (not sure if close does that) and set it to null. Otherwise if you call create again you could be calling a closed or disposed client.